### PR TITLE
ci: Refactor build functions

### PR DESCRIPTION
**Pull request by chivay**
Additionally:

+ Use Release VMI build
+ Enable VMI debugging with env variable

### CODE REVIEW

File: ci/build_utils.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    mkdir -p $(pwd)/build
+    cd build
+    cmake .. -DCMAKE_INSTALL_PREFIX=$PREFIX \
+             -DENABLE_FILE=OFF \
+             -DENABLE_LINUX=OFF \
+             -DENABLE_FREEBSD=OFF \
+             -DENABLE_KVM=OFF \
+             -DENABLE_BAREFLANK=OFF
+    make -j$(nproc)
+    make install
```
Comment (icedevml) [inline ci/build_utils.sh:24]: Looks like we are missing `--enable-vmi-debug` (is there any `cmake` equivalent?)

Comment (icedevml) [reply ci/build_utils.sh:24]: :thinking: we are also missing `--disable-safety-checks` but I don't think this should be included. Be aware that some bugs might appear that we were recently using the package compiled with `--disable-safety-checks` o_O

Comment (chivay) [reply ci/build_utils.sh]: added the `-DENV_DEBUG=ON` flag

**Review by icedevml — APPROVED** [review]
:shipit:

**Pull request merged**